pithos/GstPlayer.py

```diff
@@ -0,0 +1,506 @@
+        # or Play(an/or back again) go though all the motions but don't actaully change the
+        # player's state to the desired state until buffering has completed.
+        percent = message.parse_buffering()
+
+        if percent < 100 and self._actual_state is not PseudoGst.BUFFERING:
+            logging.debug('Buffer underrun')
+            if self._set_player_state(PseudoGst.BUFFERING):
+                logging.debug('Pausing pipeline')
+        elif percent == 100 and self._actual_state is PseudoGst.BUFFERING:
+            logging.debug('Buffer overrun')
+            if self._desired_state is PseudoGst.STOPPED:
+                if self._set_player_state(PseudoGst.PLAYING, change_gst_state=True):
+                    logging.debug('Song starting')
+            elif self._desired_state is PseudoGst.PLAYING:
+                if self._set_player_state(PseudoGst.PLAYING, change_gst_state=True):
+                    logging.debug('Restarting pipeline')
+            elif self._desired_state is PseudoGst.PAUSED:
+                if self._set_player_state(PseudoGst.PAUSED, change_gst_state=True):
+                    logging.debug('User paused')
+            # Tell everyone to update their clocks after we're done buffering or
+            # in case it takes a while after the song-changed signal for actual playback to begin.
+            self.emit('buffering-finished', self.query_position())
+
+    def _set_player_state(self, target, change_gst_state=False):
+        change_gst_state = change_gst_state or self._actual_state is not PseudoGst.BUFFERING
+        if change_gst_state:
+            ret = self.set_state(target.state)
+            if ret == Gst.StateChangeReturn.FAILURE:
+                current_state = self.state_get_name(self._actual_state.state)
+                target_state = self.state_get_name(target.state)
+                logging.warning('Error changing player state from: {} to: {}'.format(current_state, target_state))
+                return False
+            self._actual_state = target
+        if target is not PseudoGst.BUFFERING:
+            self._desired_state = target
+        if target is PseudoGst.STOPPED and self._update_proxy:
+            self._set_player_proxy()
+        self.emit('new-player-state')
+        return True
+
+    def _get_proxy(self):
+        proxy = self._settings['proxy']
+        if proxy:
+            return proxy
+        system_proxies = urllib.request.getproxies()
+        if 'http' in system_proxies:
+            return system_proxies['http']
+
+    def _set_player_proxy(self):
+        self._update_proxy = False
+        scheme, user, password, hostport = parse_proxy(self._get_proxy())
+        self.props.proxy = hostport
+        self.props.proxy_id = user
+        self.props.proxy_pw = password
+        logging.debug(
+            '\nSetting GstPlayer Proxy:'
+                '\nHTTP proxy server URI: {}'
```

Comment:
__[Flake8]__

continuation line unaligned for hanging indent


<!-- Comment by [SideCI](https://sideci.com) -->